src/sota_tools/ostree_http_repo.cc

```diff
@@ -40,23 +42,49 @@ OSTreeRef OSTreeHttpRepo::GetRef(const std::string &refname) const { return OSTr
 
 OSTreeObject::ptr OSTreeHttpRepo::GetObject(const uint8_t sha256[32]) const { return GetObject(OSTreeHash(sha256)); }
 
+bool failure = false;
+
 bool OSTreeHttpRepo::Get(const boost::filesystem::path &path) const {
+  bool ret = false;
+  CURLcode err = CURLE_OK;
   CURL *easy_handle = curl_easy_init();
   curl_easy_setopt(easy_handle, CURLOPT_VERBOSE, get_curlopt_verbose());
   server_->InjectIntoCurl(path.string(), easy_handle);
   curl_easy_setopt(easy_handle, CURLOPT_WRITEFUNCTION, &OSTreeHttpRepo::curl_handle_write);
   boost::filesystem::create_directories((root_ / path).parent_path());
-  int fp = open((root_ / path).c_str(), O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH);
+  std::string filename = (root_ / path).string();
+  int fp = open(filename.c_str(), O_WRONLY | O_CREAT, S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH);
+  if (fp == -1) {
+    LOG_ERROR << "Failed to open file: " << filename;
+    ret = false;
+    goto cleanup;
```

Comment:
@lbonn I should complain about this goto since it only saves you one indention.

Reply:
Yes, I actually like this kind of goto sometimes but since that's the only one in the C++ part of the code base, let's remove it :).

Reply:
It's true, sometimes it's useful when you've got a quadruply-nested condition or something. But I don't think this situation is bad enough to warrant it.